swift/ql/lib/codeql/swift/elements/decl/MethodDecl.qll

```diff
@@ -21,4 +21,33 @@ class MethodDecl extends AbstractFunctionDecl {
     or
     this = getAMember(any(ProtocolDecl c))
   }
+
+  /**
+   * Holds if this function is called `funcName` and its a member of a
+   * class, struct, extension, enum or protocol call `typeName`.
+   */
+  predicate hasQualifiedName(string typeName, string funcName) {
+    this.getName() = funcName and
+    (
+      exists(NominalTypeDecl c |
+        c.getFullName() = typeName and
```

Comment:
this might incur the string concatenation performance penalty, but only in case of nested types. We might maybe want to revisit this if this proves to be bothersome. Doesn't `Type::getName` already give the dotted name of nested types straight out of the DB?

Reply:
I don't think it does. It certainly didn't appear to be the case when I added it for MaD 🤔.